system/jhtree/ctfile.cpp

```diff
@@ -103,26 +104,27 @@ extern bool isCompressedIndex(const char *filename)
     if (size)
     {
         KeyHdr hdr;
-        if (io->read(0, sizeof(hdr), &hdr) == sizeof(hdr))
+        if (!io->read(0, sizeof(hdr), &hdr) == sizeof(hdr))
```

Comment:
Compare ignoring whitespace.  This change is refactoring in preparation for avoiding reading the leading header.  I could regress if undesirable.